### PR TITLE
Add x-data-checker

### PR DESCRIPTION
**Pull request by A6GibKm**
Automatically sends pull request when there are new releases.

### CODE REVIEW

File: org.bitcoincore.bitcoin-qt.json [hunk 1]
```diff
@@ -35,19 +35,37 @@
...
           "only-arches": ["arm"],
           "url": "https://bitcoincore.org/bin/bitcoin-core-22.0/bitcoin-22.0-arm-linux-gnueabihf.tar.gz",
-          "sha256": "https://bitcoincore.org/bin/bitcoin-core-22.0/bitcoin-22.0-arm-linux-gnueabihf.tar.gz"
+          "sha256": "b8713c6c5f03f5258b54e9f436e2ed6d85449aa24c2c9972f91963d413e86311",
+          "x-checker-data": {
+            "type": "anitya",
+            "project-id": "13618",
+            "version-pattern": "([\d+.]+\d+)",
+            "url-template": "https://bitcoincore.org/bin/bitcoin-core-$version/bitcoin-$version-arm-linux-gnueabihf.tar.gz"
+          }
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.bitcoincore.bitcoin-qt.json:56]: but then there is no point adding the f-e-d-c for it then. It's just a waste of everything.

**Comment by MarcoFalke** [comment]
Is there documentation available for further reading?

**Comment by A6GibKm** [comment]
> Is there documentation available for further reading?

Yes https://github.com/flathub/org.flathub.flatpak-external-data-checker, added to the commit as well.

**Comment by A6GibKm** [comment]
Updated so to use https://bitcoincore.org/en/releasesrss.xml, this should be more reliable as long as the format does not change.

**Comment by MarcoFalke** [comment]
> Yes https://github.com/flathub/flatpak-external-data-checker, added to the commit as well.

Ah ok, so this is just something for you or anyone else to run locally and it doesn't affect flathub builds at all?

**Comment by A6GibKm** [comment]
> > Yes https://github.com/flathub/flatpak-external-data-checker, added to the commit as well.
> 
> Ah ok, so this is just something for you or anyone else to run locally and it doesn't affect flathub builds at all?

On Flathub side it is run periodically, if a new release is found a PR is open it just automates one part of the job (The PR can be later reviewed and hashes checked, etc.). 

But no, Flathub builds themselves are not affected.

**Comment by MarcoFalke** [comment]
Ok, let's try it, as long as it doesn't automatically update the metadata in the background, it should be fine.

File: org.bitcoincore.bitcoin-qt.json [hunk 2]
```diff
@@ -35,13 +35,25 @@
           "type": "archive",
           "only-arches": ["x86_64"],
           "url": "https://bitcoincore.org/bin/bitcoin-core-22.0/bitcoin-22.0-x86_64-linux-gnu.tar.gz",
-          "sha256": "59ebd25dd82a51638b7a6bb914586201e67db67b919b2a1ff08925a7936d1b16"
+          "sha256": "59ebd25dd82a51638b7a6bb914586201e67db67b919b2a1ff08925a7936d1b16",
+          "x-checker-data": {
+            "type": "html",
+            "url": "https://bitcoincore.org/en/releasesrss.xml",
+            "version-pattern": "<title>Bitcoin Core ([\\d.]+)</title>",
```
Comment (MarcoFalke) [inline org.bitcoincore.bitcoin-qt.json:42]: Doesn't `.` mean wildcard? Shouldn't this be `\.` or `\\.`?

Comment (A6GibKm) [reply org.bitcoincore.bitcoin-qt.json]: Done.

**Pull request merged**